core/src/main/java/org/frankframework/jta/SpringTxManagerProxy.java

```diff
@@ -66,14 +65,13 @@ public TransactionStatus getTransaction(TransactionDefinition txDef) throws Tran
 
 	@Override
 	public void commit(TransactionStatus txStatus) throws TransactionException {
-		if (txStatus.isRollbackOnly()) {
-			Exception e = new Exception("<TX> Warning - Silent rollback from commit");
-			log.info("<TX> Executing silent rollback without exception from tx.commit! TransactionStatus: [{}], Stacktrace:", txStatus, e);
-			rollback(txStatus);
-		} else {
-			if (trace && log.isDebugEnabled()) log.debug("committing transaction [{}]", txStatus);
-			getRealTxManager().commit(txStatus);
+		if (trace && log.isDebugEnabled()) {
+			if (txStatus.isRollbackOnly())
+				log.debug("<TX> Executing rollback from tx.commit. TransactionStatus: [{}], Stacktrace:", txStatus, new Exception("<TX> Rollback from commit"));
+			else
+				log.debug("committing transaction [{}]", txStatus);
```

Comment:
The super-method will automatically do rollback, and throw an exception for flow-control so that other code does not continue as if `commit()` was successful.

What we were really accomplishing here before was to suppress proper error-handling of the `commit()`.